### PR TITLE
bpo-44220: Export PyStructSequence_UnnamedField in C API

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
The field is already present in the documentation so I think it's part of the C API:
https://docs.python.org/3/c-api/tuple.html#c.PyStructSequence_UnnamedField

https://bugs.python.org/issue44220

### CODE REVIEW

**Comment by shreyanavigyan** [comment]
I don't know but shouldn't it be "added 3.10" instead of "added 3.11 # and also 3.10"?

**Comment by Fidget-Spinner** [comment]
> I don't know but shouldn't it be "added 3.10" instead of "added 3.11 # and also 3.10"?

I'm not sure. Currently it's getting merged into the 3.11/main branch then backported to 3.10. So I thought 3.11 would be appropiate.

**Comment by shreyanavigyan** [comment]
> I'm not sure. Currently it's getting merged into the 3.11/main branch then backported to 3.10. So I thought 3.11 would be appropiate.

But when it's merge, it would seem the opposite. Backporting is for our convenience. But others will see it as 3.10 -> 3.11 not 3.11 -> 3.10.

**Comment by Fidget-Spinner** [comment]
> > I'm not sure. Currently it's getting merged into the 3.11/main branch then backported to 3.10. So I thought 3.11 would be appropiate.
> 
> But when it's merged, it would seem the opposite. Backporting is for our convenience. But others will see it as 3.10 -> 3.11 not 3.11 -> 3.10.

I'm no expert on the C API, but isn't this file an internal thing not exposed to users? I thought it's just used for tracking changes and validating exported symbols for ourselves?

There are other examples in this file of adding something in a point release and not calling it "added since that". Eg. some things were added in 3.7 and 3.6.1, but 3.7 is still used instead of saying 3.6.1 onwards (which is what is normally said in user facing whatsnew/news entry).

I don't know enough to comment further. So I'll see what Petr/another core dev says and follow that. Thanks for the concern!

File: Misc/NEWS.d/next/C API/2021-05-24-22-12-40.bpo-44220.H9CUGl.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+:c:var:`PyStructSequence_UnnamedField` is now present in ``python3.lib`` and
+``python3.dll``.
```
Comment (encukou) [inline Misc/NEWS.d/next/C API/2021-05-24-22-12-40.bpo-44220.H9CUGl.rst]:
```suggestion
:c:var:`PyStructSequence_UnnamedField` is added to the Stable ABI.
```

**Comment by encukou** [comment]
PyStructSequence_UnnamedField is already part of the C API. This commit is adding it to the *limited* API, which is a subset of the full C API.

If the PR is backported, the docs should say it was added in 3.10.

Adding something to the Stable ABI in a point release, like 3.6.1, doesn't make much sense, and it should not happen from now on. The Stable ABI doesn't really have point releases.
For the historical cases, something added in Python 3.6.1 is documented as added in Stable ABI 3.7.

**Comment by Fidget-Spinner** [comment]
> PyStructSequence_UnnamedField is already part of the C API. This commit is adding it to the *limited* API, which is a subset of the full C API.
> 
> If the PR is backported, the docs should say it was added in 3.10.
> 
> Adding something to the Stable ABI in a point release, like 3.6.1, doesn't make much sense, and it should not happen from now on. The Stable ABI doesn't really have point releases.
> For the historical cases, something added in Python 3.6.1 is documented as added in Stable ABI 3.7.

Thanks for the clarification Petr! I have a few questions:
1. Should this variable be in the limited API or public C API? I'm having second thoughts about this now.
2. Does this mean we can't change the value of the constant once its added to the limited API (or just the type)? The devguide isn't too clear about this.

Thanks in advance.

**Comment by encukou** [comment]
1. I'd say Limited API. It's not possible to mark unnamed fields using the existing Limited API, right?
2. It's possible to change the value. 
    If it was a `#define`, it wouldn't be possible to change it, since the value would be compiled into people's extensions.
    Since it's a `PyAPI_DATA`, extensions will retrieve the value by name.
    It won't be possible to change the type.

**Comment by Fidget-Spinner** [comment]
> 1. I'd say Limited API. It's not possible to mark unnamed fields using the existing Limited API, right?
> 2. It's possible to change the value.
>    If it was a `#define`, it wouldn't be possible to change it, since the value would be compiled into people's extensions.
>    Since it's a `PyAPI_DATA`, extensions will retrieve the value by name.
>    It won't be possible to change the type.

1. Yes. 
2. Thanks so much for the clarifications. This cleared up a lot of my confusion.

**Comment by vstinner** [comment]
IMO it's now too late to backport it to 3.10 which is now released. I suggest to only change the limited C API in Python 3.11.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**